### ISSUE TITLE
Ability to serially run postinstall in workspaces

### ISSUE THREAD

**Issue by jfrolich** [yarnpkg/berry#648]
- [ ] I'd be willing to implement this feature
- [ ] This feature can already be implemented through a plugin

**Describe the user story**

Bucklescript (a compiler) clashes when run in two workspace packages simultaneously because it's transforming the same source due to symlinking in workspaces. It's nice however to run it in a postinstall, because the node modules sources need to be compiled to javascript.

**Describe the solution you'd like**

A way to run postinstalls serially instead of simultaniously.

**Describe the drawbacks of your solution**

Not really a drawback, but if it's the default option, the postinstall phase might be a bit slower.

**Describe alternatives you've considered**

-

**Additional context**

Any other ideas welcome.

**Comment by arcanis** [comment]
I never used Bucklescript; can you make a repro using [Sherlock](https://next.yarnpkg.com/advanced/sherlock)?

**Comment by jfrolich** [comment]
It's non-deterministic, but `postinstall` of packages in a single workspace **may** run at the same time as it's run in parallel. Do you need a repro because it shouldn't run in parallel? Bucklescript is not super relevant, because these collisions can happen for other tools as well (if they are not resistant to working in the same directory at the same time).

**Comment by arcanis** [comment]
I don't think this is something we'll want to implement in Yarn. If a postinstall step doesn't support multiple instances running in parallel, then it should include its own mutex. There is no reason why it should be up to the consumer to know that they need their scripts to execute in a specific way for the install to work.

**Comment by arcanis** [comment]
Closing this as wontfix.

**Comment by jfrolich** [comment]
But most tools are not aware of the symlinks, so they are working with the same files.
